### PR TITLE
Add basic support for Quality scale

### PR DESCRIPTION
**Pull request by fabaff**
**Description:**
Visual representation of the new [Quality scale](https://developers.home-assistant.io/docs/en/next/integration_quality_scale_index.html).

ha-qa

To use:

```
ha_qa_scale: platinum
ha_qa_scale: gold
ha_qa_scale: silver
```

- [ ] Link needs to be updated. 

Details: https://github.com/home-assistant/architecture/issues/20

## Checklist:

- [x] Branch: `next` is for changes and new documentation that will go public with the next [home-assistant](https://github.com/home-assistant/home-assistant) release. Fixes, changes and adjustments for the current release should be created against `current`.
- [x] The documentation follow the [standards][standards].

[standards]: https://home-assistant.io/developers/documentation/standards/

### CODE REVIEW

**Comment by fabaff** [comment]
Should we link to the anchor? Thus `#gold` if the scale is gold.

File: source/_includes/asides/component_navigation.html [hunk 1]
```diff
@@ -28,6 +28,19 @@
     </div>
   {% endif %}
 
+  {% if page.ha_qa_scale %}
+    <div class='section'>
+      Quality Scale<sup><a href='https://developers.home-assistant.io/docs/en/next/integration_quality_scale_index.html'><i class="icon-info-sign"></i></a></sup>: 
+        {% if page.ha_qa_scale == 'platinum' %}
+          🏆
```
Comment (MartinHjelmare) [inline source/_includes/asides/component_navigation.html]: I think we should write out the name of the level too.

File: source/_components/sensor.random.markdown [hunk 2]
```diff
@@ -11,6 +11,7 @@ logo: home-assistant.png
 ha_category: Utility
 ha_iot_class: "Local Polling"
 ha_release: 0.32
+ha_qa_scale: silver
```
Comment (balloob) [inline source/_components/sensor.random.markdown]: I think that we should limit the QA scale to external services/integrations. Things that solely rely on internal code should be excluded.

File: source/_components/switch.mystrom.markdown [hunk 3]
```diff
@@ -11,6 +11,7 @@ logo: mystrom.png
 ha_category: Switch
 ha_release: 0.9
 ha_iot_class: "Local Polling"
+ha_qa_scale: silver
```
Comment (balloob) [inline source/_components/switch.mystrom.markdown]: Ah, this is an interesting one. I think that we should make silver level include that authentication is done at a component level, not a platform level. That will prevent breaking changes in the future.

File: source/_components/sensor.random.markdown (hunk 2 above)
Comment (fabaff) [reply source/_components/sensor.random.markdown]: This is just an example.

File: source/_components/switch.mystrom.markdown (hunk 3 above)
Comment (fabaff) [reply source/_components/switch.mystrom.markdown]: This as well. I will remove both when the link is updated. Sorry, for the confusion.

**Comment by fabaff** [comment]
I guess that the screenshot shows it. No need for live examples.

File: source/_includes/asides/component_navigation.html [hunk 4]
```diff
@@ -28,6 +28,19 @@
     </div>
   {% endif %}
 
+  {% if page.ha_qa_scale %}
+    <div class='section'>
+      Quality Scale: 
+        {% if page.ha_qa_scale == 'platinum' %}
+          <a href='https://developers.home-assistant.io/docs/en/next/integration_quality_scale_index.html#platinum'>🏆</a>
```
Comment (balloob) [inline source/_includes/asides/component_navigation.html]: I think that we should add a page about the Integration Quality Scale to the normal documentation and make that user focused. It should not include the requirements but just the descriptions of the different levels.

Comment (fabaff) [reply source/_includes/asides/component_navigation.html]: That's a valid point.

Comment (balloob) [reply source/_includes/asides/component_navigation.html]: We can just start with a page in the docs that just mentions the existing descriptions from the dev site.

Comment (balloob) [reply source/_includes/asides/component_navigation.html]: This link still needs updating.

File: source/_docs/quality_scale.markdown [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+This integration passes the bare minimum requirements. That's the level of most integrations when they are introduced into Home Assistant. It doesn't mean that they are bad or buggy, just that you need to configure them with an entry in your `configuration.yaml` file.
+
+## {% linkable_title Silver 🥈 %}
+
+This integration is able to cope when things go wrong. It will not print any exceptions nor will it fill the log with retry attempts. Also, it will show you if a device is offline or not ready when you start Home Assistant.
+
+## {% linkable_title Gold 🥇 %}
+
+This is a solid integration that is able to survive poor conditions and can be configured via the user interface. It may contain translated states and is covered by unit tests.
```
Comment (balloob) [inline source/_docs/quality_scale.markdown]: Let's remove the line about translated states and unit tests. Not too interesting for user.

File: source/_includes/asides/component_navigation.html (hunk 1 above)
Comment (balloob) [reply source/_includes/asides/component_navigation.html]: Yeah let's do that.

**Review by balloob — DISMISSED** [review]
Ok to merge when name of levels added to emoji's. Not all platforms can render emoji's perfectly.

**Review by balloob — APPROVED** [review]

**Pull request merged**